app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -37,13 +38,14 @@ module SmartProxyExtensions
         has_many :lifecycle_environments,
                  :class_name => "Katello::KTEnvironment",
                  :through    => :capsule_lifecycle_environments,
-                 :source     => :lifecycle_environment
+                 :source     => :lifecycle_environment,
+                 :dependent  => :nullify
```

Comment:
The rest of these look fine, this one i'm still trying to figure out what :nullify would even do here.  It seems like we'd want 'no action', since the has_many :capsule_lifecycle_environments would handle the destroy of the join object

Reply:
actually, this isn't reported through rubocop, and looking at https://github.com/bbatsov/rubocop/issues/4751 it is not expected to, so i think has_many :through associations should still not have a :dependent setting

Reply:
You're right, it was just for the `has_many :hostgroups` below, updating now!